lib/OpenQA/Utils.pm

```diff
@@ -68,8 +92,9 @@ sub data_name($) {
     return $1;
 }
 
-sub needledir($$) {
-    return testcasedir($_[0], $_[1]) . '/needles';
+sub needledir {
+    my ($distri, $version, $test_variant) = @_;
```

Comment:
I have to ask again: what is the significance of that 3rd parameter. I don't see it called like this ever